app/renderer/js/main.ts

```diff
@@ -235,6 +238,20 @@ export class ServerManagerView {
   initSidebar(): void {
     const showSidebar = ConfigUtil.getConfigItem("showSidebar", true);
     this.toggleSidebar(showSidebar);
+    this.sortableSidebar = new SortableJS(this.$tabsContainer, {
+      animation: 150,
+      onEnd: (event: SortableJS.SortableEvent) => {
+        // Update the domain order in the database
+        DomainUtil.updateDomainOrder(event.oldIndex ?? 0, event.newIndex ?? 0);
```

Comment:
Unexpected data should be rejected and ignored, not silently replaced with fake values like `0`.